Odata-docs/webapi-8/routing-overview.md

```diff
@@ -0,0 +1,365 @@
+
+var modelBuilder = new ODataConventionModelBuilder();
+modelBuilder.EntitySet<Customer>("Customers");
+
+builder.Services.AddControllers().AddOData(
+    options => options.EnableQueryFeatures(maxTopValue: null).AddRouteComponents(
+        routePrefix: "odata",
+        model: modelBuilder.GetEdmModel()));
+
+var app = builder.Build();
+```
+
+# [.NET Core 3.1](#tab/netcoreapp31)
+
+```csharp
+// Startup.cs
+public void ConfigureServices(IServiceCollection services)
+{
+    var modelBuilder = new ODataConventionModelBuilder();
+    modelBuilder.EntitySet<Customer>("Customers");
+
+    services.AddControllers().AddOData(
+        options => options.EnableQueryFeatures(maxTopValue: null).AddRouteComponents(
+            routePrefix: "odata",
+            model: modelBuilder.GetEdmModel()));
+}
+```
+
+--
+
+## OData routing
+In the above block of code, we build an Edm model using the `ODataConventionModelBuilder`. We then call the `AddOData()` method resulting into the following OData routing conventions being registered in the [dependency injection](/aspnet/core/fundamentals/dependency-injection) container:
+- `MetadataRoutingConvention` - routes requests for service metadata
+- [`EntitySetRoutingConvention`](/odata/webapi-8/entityset-routing) - routes requests for entity sets
+- [`EntityRoutingConvention`](/odata/webapi-8/entity-routing) - routes requests for single entities by key
+- [`SingletonRoutingConvention`](/odata/webapi-8/singleton-routing) - routes requests for singletons
+- [`FunctionRoutingConvention`](/odata/webapi-8/function-routing) - routes requests for Edm functions
+- [`ActionRoutingConvention`](/odata/webapi-8/action-routing) - routes requests for Edm actions
+- `OperationImportRoutingConvention` - routes requests for operation imports
+- `PropertyRoutingConvention` - routes requests for structural properties
+- [`NavigationRoutingConvention`](/odata/webapi-8/navigation-routing) - routes requests for navigation properties
+- [`RefRoutingConvention`](/odata/webapi-8/ref-routing) - routes requests for entity reference links
+- `AttributeRoutingConvention` - represents a routing convention for attribute-routed requests
+
+A in-depth walkthrough for the core routing conventions can be found in the relevant tutorial.
+
+At application startup, OData path templates for all potential endpoints are built. The OData routing conventions above are "executed" to generate a route table that maps each controller action (endpoint) to an OData route template.
+
+The built-in OData routing conventions above implement the following interface:
+```csharp
+public interface IODataControllerActionConvention
+{
+    int Order { get; }
+    bool AppliesToController(ODataControllerActionContext context);
+    bool AppliesToAction(ODataControllerActionContext context);
+}
+```
+
+- `Order` is used to specify the execution order of the convention. Conventions are executed in ascending numeric value of the `Order` property.
```

Comment:
is it worth the mentioning the order of the various conventions here?

Reply:
Already, talking about `AppliesToAction`, `AppliesToController` and `Order` felt like it was making the doc too complex. But I needed to provide a big of info about what happens in the background. I think list the order of various conventions will only add to the complexity.